src/Phinx/Db/Adapter/SQLiteAdapter.php

```diff
@@ -969,7 +1127,12 @@ protected function getRenameColumnInstructions(string $tableName, string $column
             return $newState + $state;
         });
 
-        return $this->copyAndDropTmpTable($instructions, $tableName);
+        $instructions = $this->bufferIndicesAndTriggers($instructions, $tableName);
+        $instructions = $this->updateIndicesForRenamedColumn($instructions, $columnName, $newColumnName);
+        $instructions = $this->copyAndDropTmpTable($instructions, $tableName);
```

Comment:
Thoughts on modifying `copyAndDropTmpTable` to accept an optional third and fourth arguments `$columnName` and `newColumnName`, and then `copyAndDropTmpTable` is responsible for calling `bufferIndicesAndTriggers`, `updateIndicesForRenamedColumn`, and `recreateIndicesAndTriggers`?

This would cut down on the boilerplate around using these as three of them are always used in conjunction.

Reply:
I'm very much open to some refactoring here, I don't really like all that repitition either. I actually had it like this at first, with the new methods added around the copy/delete post step:

```
copyAndDropTmpTable(AlterInstructions $instructions, string $tableName, string $deletedOrRenamedColumnName, string $newColumnName)
```

It was kinda bloated, and I thought that it wouldn't be overly appreciated.

How about a new method that combines all the post step addition methods, something like `endAlterByCopyTable` to complement `beginAlterByCopyTable`?

Reply:
Sure, I'd be open to having a complimentary `endAlterByCopyTable` that wraps these 4 function calls as-is, makes sense as we have a `beginAlterByCopyTable`.